src/main/java/org/openrewrite/java/migrate/javax/AddTransientAnnotationToPrivateAccessor.java

```diff
@@ -0,0 +1,97 @@
+import org.openrewrite.java.TreeVisitingPrinter;
+import org.openrewrite.java.search.FindAnnotations;
+import org.openrewrite.java.search.UsesField;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class AddTransientAnnotationToPrivateAccessor extends Recipe {
+
+    @Override
+    public String getDisplayName() {
+        return "Private accessor methods must have a `@Transient` annotation";
+    }
+
+    @Override
+    public String getDescription() {
+        return "According to the JPA 2.1 specification, when property access is used, the property accessor methods " +
+               "must be public or protected. OpenJPA ignores any private accessor methods, whereas EclipseLink persists " +
+               "those attributes. To ignore private accessor methods in EclipseLink, the methods must have a " +
+               "`@Transient` annotation.";
+    }
+
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(
+                new UsesType<>("javax.persistence.Entity", true),
+                new JavaIsoVisitor<ExecutionContext>() {
+                    @Override
+                    public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration method, ExecutionContext ctx) {
+                        // TODO: is this needed to filter to JPA classes?
+                        // Exit if parent class is not tagged for JPA
+                        J.ClassDeclaration parentClass = getCursor().dropParentUntil(parent -> parent instanceof J.ClassDeclaration).getValue();
+                        if (FindAnnotations.find(parentClass, "javax.persistence.Entity").isEmpty()) {
+                            return method;
+                        }
+                        // Exit if not private accessor method
+                        if (!isPrivateAccessorMethod(method)) {
+                            return method;
+                        }
+                        // Add @Transient annotation
+                        // TODO: why is this running twice? I have the javaParser specifying the jar
+                        maybeAddImport("javax.persistence.Transient");
+                        return JavaTemplate.builder("@Transient")
+                                .contextSensitive()
+                                .javaParser(JavaParser.fromJavaVersion().classpathFromResources(ctx, "javax.persistence-api-2.2"))
+                                .imports("javax.persistence.Transient")
+                                .build()
+                                .apply(getCursor(), method.getCoordinates().addAnnotation(Comparator.comparing(J.Annotation::getSimpleName)));
```

Comment:
Not sure why this is getting applied twice. Last time this happened was because I was [missing the javaParser](https://github.com/openrewrite/rewrite-migrate-java/pull/410/commits/4d9e6e110d2a3ac53939b61e5d92350cd187e910) with the relevant jar, but it's already been specified here.